app/src/main/java/com/github/orkest/Model/AudioRecognition.kt

```diff
@@ -0,0 +1,174 @@
+package com.github.orkest.Model
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import androidx.annotation.RequiresPermission
+import com.github.orkest.Constants
+import com.shazam.shazamkit.AudioSampleRateInHz
+import com.shazam.shazamkit.MatchResult
+import com.shazam.shazamkit.ShazamKit
+import com.shazam.shazamkit.ShazamKitResult
+import com.shazam.shazamkit.SignatureGenerator
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import java.nio.ByteBuffer
+import java.util.concurrent.CompletableFuture
+
+class AudioRecognition {
+
+    companion object {
+
+         val catalog =
+                ShazamKit.createCustomCatalog().apply {
+
+                }
+
+        private fun Int.toByteAllocation(): Int {
+            return when (this) {
+                AudioFormat.ENCODING_PCM_16BIT -> 2
+                else -> throw IllegalArgumentException("Unsupported encoding")
+            }
+        }
+
+        private fun ByteBuffer.putTrimming(byteArray: ByteArray) {
+            if (byteArray.size <= this.capacity() - this.position()) {
+                this.put(byteArray)
+            } else {
+                this.put(byteArray, 0, this.capacity() - this.position())
+            }
+        }
+
+        /**
+         * Records audio for the maximum query duration and returns an [AudioChunk] with the recorded audio
+         * @return [AudioChunk] with the recorded audio
+         * @throws [SecurityException] if the app does not have the RECORD_AUDIO permission
+         */
+        @RequiresPermission(Manifest.permission.RECORD_AUDIO)
+        private fun audioChunkRecord(): AudioChunk {
+            val audioSource = MediaRecorder.AudioSource.UNPROCESSED
+
+            val audioFormat = AudioFormat.Builder()
+                .setChannelMask(AudioFormat.CHANNEL_IN_MONO)
+                .setEncoding(AudioFormat.ENCODING_PCM_16BIT)
+                .setSampleRate(48_000)
+                .build()
+
+            val audioRecord = AudioRecord.Builder()
+                .setAudioSource(audioSource)
+                .setAudioFormat(audioFormat)
+                .build()
+
+            val seconds = catalog.maximumQuerySignatureDurationInMs
+
+            // Final desired buffer size to allocate necessary seconds of audio
+            val size = audioFormat.sampleRate * audioFormat.encoding.toByteAllocation() * seconds
+            val destination = ByteBuffer.allocate(size.toInt())
+
+            // Small buffer to retrieve chunks of audio
+            val bufferSize = AudioRecord.getMinBufferSize(
+                48_000,
```

Comment:
Maybe we would like to modify this value later ? Can be nice to give it as a parameter